src/openms/source/ANALYSIS/MAPMATCHING/FeatureGroupingAlgorithm.cpp

```diff
@@ -129,11 +129,32 @@ namespace OpenMS
         }
       }
       *cons_it = adjusted;
+
+      for (auto& id : cons_it->getPeptideIdentifications())
+      {
+        Size file_index = id.getMetaValue("map_index");
```

Comment:
can we be sure we have a map_index here?

Reply:
That is what I am trying to enforce. ConsensusXMLs without map_index in the IDs do not make sense.

Reply:
I will add a warning but I dont know how to proceed with unannotated IDs.